### PR TITLE
LPD-25860 Checkout widget display Full Addresses and Phone Number toggle added

### PR DESCRIPTION
**Pull request by brbalazs**
https://liferay.atlassian.net/browse/LPD-25860

### CODE REVIEW

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:sf

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -12741,6 +12742,8 @@ order-status-update-interval-minutes=Order Status Update Interval (Minutes)
 order-status-update-max-interval-minutes=Order Status Update Max Interval (Minutes)
 order-submitted=Order Submitted
 order-summary=Order Summary
+order-summary-show-full-address=Order Summary - Show Full Address
+order-summary-show-phone-number=Order Summary - Show Phone Number
```
Comment (smottal) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:12746]: please remove the -

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/order_summary.jsp [hunk 2]
```diff
@@ -421,6 +422,19 @@ Map<Long, List<CommerceOrderValidatorResult>> commerceOrderValidatorResultsMap =
...
 
+					<c:if test="<%= Validator.isNotNull(shippingAddress.getZip()) && checkoutDisplayContext.isOrderSummaryShowFullAddressEnabled() %>">
+						<%= HtmlUtil.escape(shippingAddress.getZip()) %> <br />
+					</c:if>
+
+					<c:if test="<%= (shippingAddress.getRegion() != null) && checkoutDisplayContext.isOrderSummaryShowFullAddressEnabled() %>">
+
+						<%
+						Region shippingRegion = shippingAddress.getRegion();
+						%>
```
Comment (smottal) [inline modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/order_summary.jsp:433]: can you move this before the c:if so we can use the same variable also in the ` != null` statement?

File: modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/order_summary.jsp [hunk 3]
```diff
@@ -421,6 +422,19 @@ Map<Long, List<CommerceOrderValidatorResult>> commerceOrderValidatorResultsMap =
...
 					<%= HtmlUtil.escape(shippingAddress.getCity()) %> <br />
 
+					<c:if test="<%= Validator.isNotNull(shippingAddress.getZip()) && checkoutDisplayContext.isOrderSummaryShowFullAddressEnabled() %>">
+						<%= HtmlUtil.escape(shippingAddress.getZip()) %> <br />
+					</c:if>
+
+					<c:if test="<%= (shippingAddress.getRegion() != null) && checkoutDisplayContext.isOrderSummaryShowFullAddressEnabled() %>">
+
+						<%
+						Region shippingRegion = shippingAddress.getRegion();
```
Comment (smottal) [inline modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/order_summary.jsp:432]: I see that there is also a Country variable that is used for shipping and billing without conflicts. Can we try to call this only region?

File: modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/order_summary.jsp [hunk 4]
```diff
@@ -497,13 +513,30 @@ Map<Long, List<CommerceOrderValidatorResult>> commerceOrderValidatorResultsMap =
...
 
+					<c:if test="<%= Validator.isNotNull(commerceBillingAddress.getZip()) && checkoutDisplayContext.isOrderSummaryShowFullAddressEnabled() %>">
+						<%= HtmlUtil.escape(commerceBillingAddress.getZip()) %> <br />
+					</c:if>
+
+					<c:if test="<%= (commerceBillingAddress.getRegion() != null) && checkoutDisplayContext.isOrderSummaryShowFullAddressEnabled() %>">
+
+						<%
+						Region billingRegion = commerceBillingAddress.getRegion();
+						%>
```
Comment (smottal) [inline modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/order_summary.jsp:524]: move this before the c:if statement and try to call it only region

File: modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/portlet/action/CommerceCheckoutConfigurationAction.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+		if (cmd.equals(Constants.UPDATE)) {
+			PortletPreferences portletPreferences =
+				actionRequest.getPreferences();
+
+			portletPreferences.setValue(
+				"showCommerceCheckoutFullAddress",
+				getParameter(actionRequest, "showCommerceCheckoutFullAddress"));
+			portletPreferences.setValue(
+				"showCommerceCheckoutPhoneNumber",
+				getParameter(actionRequest, "showCommerceCheckoutPhoneNumber"));
```
Comment (smottal) [inline modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/portlet/action/CommerceCheckoutConfigurationAction.java:95]: I think these should use the new name. (I don't know how they can work like this becase the save of the configuration should post different names) 🤔

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by brbalazs** [comment]
ci:test:sf

**Review by smottal — APPROVED** [review]

**Comment by brbalazs** [comment]
ci:test:relevant

**Review by luca-pellizzon — APPROVED** [review]

**Pull request closed without merging**